### PR TITLE
Fixed a bug that a file is being written by more than one processor simultaneously

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
This PR is addressing issue #2228. When a file is written at different time step, it may lead to crash because different processes may try to write to it simultaneously. 
## Description

The captain process keeps a record of the files being written. If a request to write a file which is still in the record, it waits until the completion of that process. To accomplish this, the front end now sends the file name to the captain when asking for a writer. The writer sends back the file name to the captain when it is free.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (affects only documentation or cleanup)

## Checklist:

- [x] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [x] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

**Comment by weiyuan-jiang** [comment]
@bena-nasa Could you please help to test my change?

**Comment by bena-nasa** [comment]
> @bena-nasa Could you please help to test my change?

Yes, I will give this a workout

**Comment by mathomp4** [comment]
@weiyuan-jiang Note that when you are happy with this code, it probably should be re-done as a hotfix against `main` rather than a bugfix against `develop`. Since this is the sort of fix that might need to be backported to older tags, it's probably good to have it isolated from all the other work on develop.

Plus as @tclune would say, it's a bug on main in released code so it's a hotfix!

File: .github/workflows/enforce-labels.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
   require-label:
     runs-on: ubuntu-latest
     steps:
-      - uses: mheap/github-action-required-labels@v4
+      - uses: mheap/github-action-required-labels@v5
```
Comment (tclune) [inline .github/workflows/enforce-labels.yml:11]: Something is amiss. This file should not have been touched.  I fear you started with the wrong baseline.

**Comment by tclune** [comment]
This PR started from an incorrect branch.   Need to start over from fresh copy off `main` branch.

**Review by tclune — CHANGES_REQUESTED** [review]
Start from `main` not `develop`

**Pull request closed without merging**